### PR TITLE
Unify make commands for deployment

### PR DESCRIPTION
**Pull request by gkrenn**
## Description

Currently multiply options / make commands to install the operator exist.
This PR aligns them by: 
- removing all commands for deployment but `make [un]deploy`
- use helm upgrade to get it as close to customer experience as possible

[Jira](https://dt-rnd.atlassian.net/browse/DAQ-1623)

## How can this be tested?

- run `make [un]deploy`

### CODE REVIEW

File: hack/make/deploy/deploy.mk [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
-			--set platform=$(PLATFORM) \
 			--set csidriver.enabled=$(ENABLE_CSI) \
 			--set manifests=true \
-			--set image="$(IMAGE_URI)" | kubectl apply -f -
+			--set image="$(IMAGE_URI)" \
+			--set debug=$(DEBUG)
+
+## Undeploy the current operator installation
+undeploy:
+	kubectl delete dynakube --all -n dynatrace
```
[the remark this answers is not in the document]
Comment (gkrenn) [reply hack/make/deploy/deploy.mk:23]: True, we also missed that in the old command, update it 👍

File: HACKING.md [hunk 2]
```diff
@@ -7,19 +7,7 @@
...
-```sh
-make deploy/kubernetes
-```
-
-#### OpenShift
-
-```sh
-make deploy/openshift
-```
+There are automatic builds from the master branch. The latest development build can be installed using `make deploy`.
```
Comment (andriisoldatenko) [inline HACKING.md:10]:
```suggestion
There are automatic builds from the main branch. The latest development build can be installed using `make deploy`.
```

**Review by andriisoldatenko — DISMISSED** [review]
LGTM, except typo ( old one).

**Review by 0sewa0 — COMMENTED** [review]
looks good

my only question: do we know if the asperitas pipelines use any of these in their pipelines

File: HACKING.md (hunk 2 above)
Comment (gkrenn) [reply HACKING.md]: Done

**Comment by gkrenn** [comment]
Checked with asperitas, they use the helm command directly to install the operator

**Review by andriisoldatenko — APPROVED** [review]

**Review by 0sewa0 — DISMISSED** [review]

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**